### ISSUE TITLE
Possible inference bug on master with misinferred type

### ISSUE THREAD

**Issue by KristofferC** [JuliaLang/julia#34246]
The code below gives an error on Master

```jl
julia> using MarketData

julia> using TimeSeries

julia> f() = cl .< op
f (generic function with 1 method)

julia> f()
ERROR: MethodError: no method matching view(::TimeArray{Float64,1,Date,Array{Float64,1}}, ::Array{Int64,1})
Closest candidates are:
  view(::AbstractArray, ::Any...) where N at subarray.jl:161
Stacktrace:
 [1] copy(::Base.Broadcast.Broadcasted{TimeSeries.TimeArrayStyle{1},Nothing,typeof(<),Tuple{TimeArray{Float64,1,Date,Array{Float64,1}},TimeArray{Float64,1,Date,Array{Float64,1}}}}) at /Users/kristoffer/JuliaPkgEval/TimeSeries.jl/src/broadcast.jl:48
 [2] materialize at ./broadcast.jl:819 [inlined]
 [3] f() at ./REPL[33]:1
 [4] top-level scope at REPL[34]:1
```

Trying to run it through the debugger works fine:

```
julia> using Debugger

julia> @run f()
500×1 TimeArray{Bool,1,Date,BitArray{1}} 2000-01-03 to 2001-12-31
│            │ Close_Open │
├────────────┼────────────┤
```

The place where the error happens is https://github.com/JuliaStats/TimeSeries.jl/blob/c730d62e22156f5f2ec1cf4af7c4dba84114e876/src/broadcast.jl#L48-L51, with the code:

```
    TimeArray(view(timestamp(tas[1]), tstamp_idx[1]),
              broadcast(bc′.f, args...),
              col′,
              meta′)
```

Rewriting this as:

```
    t = timestamp(tas[1])
    @show typeof(t)
    @show typeof(tstamp_idx[1])
    a = view(t, tstamp_idx[1])
```

gives 

```
typeof(t) = Array{Date,1}
typeof(tstamp_idx[1]) = Array{Int64,1}
ERROR: MethodError: no method matching view(::TimeArray{Float64,1,Date,Array{Float64,1}}, ::Array{Int64,1})
```

which is a strange error message since we already established that `t` is a `Array{Date,1}` and not a `TimeArray` from the `@show`.

Changing the last line to 

```
    a = view(t, [1,2,3])
```

makes the call not error (but this shouldn't have had an effect since the second argument already was a `Vector{Int}`).

**Comment by JeffBezanson** [comment]
This looks like another instance of #34206. The type is inferred correctly, it's just getting the wrong field.

**Comment by JeffBezanson** [comment]
Fixed by #34243
